test/CppTests/Tests/BreakpointTests.cs

```diff
@@ -424,6 +424,73 @@ public void ConditionalStringBreakpoints(ITestSettings settings)
             }
         }
 
+        [Theory]
+        [DependsOnTest(nameof(CompileKitchenSinkForBreakpointTests))]
+        [RequiresTestSettings]
+        // lldb-mi does not support -break-watch
+        [UnsupportedDebugger(SupportedDebugger.Lldb, SupportedArchitecture.x64 | SupportedArchitecture.x86)]
+        public void DataBreakpointTest(ITestSettings settings)
+        {
+            this.TestPurpose("Tests that data breakpoints work");
+            this.WriteSettings(settings);
+
+            IDebuggee debuggee = SinkHelper.Open(this, settings.CompilerSettings, DebuggeeMonikers.KitchenSink.Breakpoint);
+
+            using (IDebuggerRunner runner = CreateDebugAdapterRunner(settings))
+            {
+                runner.Launch(settings.DebuggerSettings, debuggee, "-fCalling");
+
+                SourceBreakpoints callingBreakpoints = debuggee.Breakpoints(SinkHelper.Calling, 15);
+                runner.SetBreakpoints(callingBreakpoints);
+
+                runner.Expects.HitBreakpointEvent(SinkHelper.Calling, 15)
+                            .AfterConfigurationDone();
+
+                using (IThreadInspector inspector = runner.GetThreadInspector())
+                {
+                    IFrameInspector mainFrame = inspector.Stack.First();
+                    mainFrame.GetVariable("total");
+
+                    this.Comment("Get DataBreakpointInfo on 'total'");
+                    var response = runner.DataBreakpointInfo("total");
+                    Assert.NotNull(response?.body);
+                    Assert.Contains("write", response.body.accessTypes); // Validate access type is "write"
+                    Assert.Equal("When 'total' changes (8 bytes)", response.body.description, true, true, true); // Validate description matches DataBreakpointDisplayString
+                    Assert.False(string.IsNullOrEmpty(response.body.dataId));
+                    Assert.EndsWith("total,8", response.body.dataId); // Validate dataId matches format <Address>,<Id>,<Size>
+
+                    this.Comment("SetDataBreakpoint on 'total' Info");
+                    DataBreakpoints dataBreakpoints = new DataBreakpoints();
+                    dataBreakpoints.Add(response.body.dataId);
+                    runner.SetDataBreakpoints(dataBreakpoints);
+                }
+
+                this.Comment("Run to statement after data breakpoint");
+                // Note this is going to be source line 15 for `i++`.
+                runner.Expects.HitBreakpointEvent(SinkHelper.Calling, 15)
+                              .AfterContinue();
+
+                using (IThreadInspector inspector = runner.GetThreadInspector())
+                {
+                    IFrameInspector mainFrame = inspector.Stack.First();
+                    Assert.Equal("1.0", mainFrame.GetVariable("total").Value);
```

Comment:
GDB seems to be evil and returning `value` as `1` for type `double`.
```
  Failed CppTests.Tests.BreakpointTests.DataBreakpointTest(settings: GppMingw64/x64/GdbMingw64) [3 s]
  Error Message:
   Assert.Equal() Failure
           ↓ (pos 1)
Expected: 1.0
Actual:   1
           ↑ (pos 1)
```

Reply:
`double.TryParse`?